README.rst

```diff
@@ -110,147 +145,237 @@ version of the AWS Encryption SDK, we recommend using the default value.
 
 Encryption and Decryption
 =========================
-After you create an instance of an ``EncryptionSDKClient`` and a ``MasterKeyProvider``, you can use either of
+After you create an instance of an ``EncryptionSDKClient`` and a ``Keyring``, you can use either of
 the client's two ``encrypt``/``decrypt`` functions to encrypt and decrypt your data.
 
-.. code:: python
-
-    import aws_encryption_sdk
-    from aws_encryption_sdk.identifiers import CommitmentPolicy
-
-    client = aws_encryption_sdk.EncryptionSDKClient(
-        commitment_policy=CommitmentPolicy.FORBID_ENCRYPT_ALLOW_DECRYPT
-    )
+Here's an example for using a KMS keyring for encryption and decryption:
 
-    kms_key_provider = aws_encryption_sdk.StrictAwsKmsMasterKeyProvider(key_ids=[
-        'arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
-        'arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333'
-    ])
-    my_plaintext = b'This is some super secret data!  Yup, sure is!'
+.. code:: python
 
-    my_ciphertext, encryptor_header = client.encrypt(
+    # Encrypt the data.
+    my_ciphertext, enc_header = client.encrypt(
         source=my_plaintext,
-        key_provider=kms_key_provider
+        keyring=kms_keyring
     )
 
-    decrypted_plaintext, decryptor_header = client.decrypt(
+    # Decrypt your encrypted data.
+    my_decrypted_plaintext, dec_header = client.decrypt(
         source=my_ciphertext,
-        key_provider=kms_key_provider
+        keyring=kms_keyring
     )
 
-    assert my_plaintext == decrypted_plaintext
-    assert encryptor_header.encryption_context == decryptor_header.encryption_context
-
 You can provide an `encryption context`_: a form of additional authenticating information.
 
+Here's an example for using KMS keyring with an encryption context:
+
 .. code:: python
 
+    import boto3
+    from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+    from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+    from aws_cryptographic_materialproviders.mpl.models import CreateAwsKmsKeyringInput
+    from aws_cryptographic_materialproviders.mpl.references import IKeyring
+
     import aws_encryption_sdk
-    from aws_encryption_sdk.identifiers import CommitmentPolicy
+    from aws_encryption_sdk import CommitmentPolicy
 
+    # Instantiate the encryption SDK client.
     client = aws_encryption_sdk.EncryptionSDKClient(
-        commitment_policy=CommitmentPolicy.FORBID_ENCRYPT_ALLOW_DECRYPT
+        commitment_policy=CommitmentPolicy.REQUIRE_ENCRYPT_REQUIRE_DECRYPT
+    )
+
+    # Create an encryption context
+    encryption_context: Dict[str, str] = {
+        "encryption": "context",
+        "is not": "secret",
+        "but adds": "useful metadata",
+        "that can help you": "be confident that",
+        "the data you are handling": "is what you think it is",
+    }
+
+    # Create a KMS keyring.
```

Comment:
Keep "Create KMS keyring" in! I think these comments are helpful, since creating a keyring is a multi-step process.

Reply:
I think since we already explain a 'create a KMS Multi Keyring' step, in order to keep it concise we shouldn't add a "Create KMS keyring" explanation here.